src/Cli/dotnet/Commands/Test/SolutionAndProjectUtility.cs

```diff
@@ -230,6 +230,17 @@ public static IEnumerable<ParallelizableTestModuleGroupWithSequentialInnerModule
 
         string targetFramework = project.GetPropertyValue(ProjectProperties.TargetFramework);
         RunProperties runProperties = GetRunProperties(project, loggers);
+        if (string.IsNullOrEmpty(runProperties.RunCommand) || runProperties.RunCommand.HasExtension(CliConstants.DLLExtension))
```

Comment:
If `RunCommand` is empty, `dotnet run` already throws an exception. So I'm doing the same.

I'm also throwing if `RunCommand` is set to dll. But this should never happen in any "default" scenario unless the user is really setting `RunCommand` themselves.